main.tf

```
@@ -1,13 +1,5 @@
-# fail-fast: ensure the project exists at plan time and that we have access
-data "google_project" "project" {
-  count = var.module_enabled ? 1 : 0
-
-  project_id = var.project
-}
-
 locals {
-  project           = try(data.google_project.project[0].project_id, "")
-  precomputed_email = "${var.account_id}@${local.project}.iam.gserviceaccount.com"
+  precomputed_email = "${var.account_id}@${var.project}.iam.gserviceaccount.com"
```

we can only pre-compute when project is present.. 

we should discuss making project required everywhere as falling back to providers is always bad..

```suggestion
  precomputed_email = try("${var.account_id}@${var.project}.iam.gserviceaccount.com", null)
```